### ISSUE TITLE
Weird progress indicators started appearing in stdout

### ISSUE THREAD

**Issue by martinstenhoff** [dotnet/sdk#44610]
Why have I started seeing the following character sequence (which seems to some progress indicator) whenever I pipe dotnet program output to a file now?

```\x1b]9;4;3;\x1b\\\x1b]9;4;0;\x1b\\```

Program.cs:
```
Console.WriteLine("Hello, World!");
```

dotnet run > output.txt

output.txt:
```
]9;4;3;\]9;4;0;\Hello, World!
```

```
❯ dotnet --version
9.0.100-rc.2.24474.11
```

OS: Windows 11

**Comment by martinstenhoff** [comment]
This issue talks about some new console logger in .Net9
https://github.com/dotnet/msbuild/issues/9950

I still see these characters in the output file if build and run are done in separate steps:
```
dotnet build -c Release
dotnet run -c Release --no-build -v q > .\output.txt
```

And it happens in both Command Prompt and Powershell terminal

**Comment by martinstenhoff** [comment]
@KalleOlaviNiemitalo thanks for the investigation :)

Seems like this has been raised before and we'll have to live with this for some time after GA :(
https://github.com/dotnet/msbuild/issues/10841
https://github.com/dotnet/msbuild/issues/10579

**Comment by MD-V** [comment]
This also currently breaks .NET custom providers in pulumi:

`I1113 09:01:30.634581   44896 log.go:73] Got plugin response: stdout:"\x1b]9;4;3;\x1b\\"
I1113 09:01:30.634581   44896 log.go:73] Waiting for plugin message
I1113 09:01:30.636850   44896 log.go:73] Got plugin response: stdout:"\x1b]9;4;0;\x1b\\"
I1113 09:01:30.636850   44896 log.go:73] Waiting for plugin message
I1113 09:01:30.818760   44896 log.go:73] Got plugin response: stdout:"61683\n"
I1113 09:01:30.818760   44896 log.go:73] Waiting for plugin message
I1113 09:01:30.818760   44896 deployment_executor.go:256] deploymentExecutor.Execute(...): error handling event: could not load provider for resource urn:pulumi:engineering_app_dev_1::engineering_app_azure::internalprovider:resources:appsettings::apiAppSettings: could not create provider urn:pulumi:engineering_app_dev_1::engineering_app_azure::pulumi:providers:internalprovider::default::effc6bd2-4b36-4c51-9b0d-175fc9573a81: load plugin for internalprovider provider 'urn:pulumi:engineering_app_dev_1::engineering_app_azure::pulumi:providers:internalprovider::default': internalprovider (resource) plugin [C:\Projects\Projekte\Engineering\Infrastructure\Pulumi\internalprovider\pulumi-resource-internalprovider.exe] wrote a non-numeric port to stdout ('0'): strconv.Atoi: parsing "\x1b]9;4;3;\x1b\\\x1b]9;4;0;\x1b\\61683": invalid syntax
I1113 09:01:30.818760   44896 eventsink.go:86] eventSink::Error(<{%reset%}>could not load provider for resource urn:pulumi:engineering_app_dev_1::engineering_app_azure::internalprovider:resources:appsettings::apiAppSettings: could not create provider urn:pulumi:engineering_app_dev_1::engineering_app_azure::pulumi:providers:internalprovider::default::effc6bd2-4b36-4c51-9b0d-175fc9573a81: load plugin for internalprovider provider 'urn:pulumi:engineering_app_dev_1::engineering_app_azure::pulumi:providers:internalprovider::default': internalprovider (resource) plugin [C:\Projects\Projekte\Engineering\Infrastructure\Pulumi\internalprovider\pulumi-resource-internalprovider.exe] wrote a non-numeric port to stdout ('0'): strconv.Atoi: parsing "\x1b]9;4;3;\x1b\\\x1b]9;4;0;\x1b\\61683": invalid syntax<{%reset%}>)
`

**Comment by sailro** [comment]
We have a host application, running .net programs. For easier debugging we do the following:
- run directly the compiled .net program (when running in production)
- use dotnet run --project (when running in "dev mode")

And of course those programs are using the console to exchange data with the host.

So when upgrading to net 9, happily writing escape codes in the console without any switch to disable this new behavior, everything is broken. 💔 

![Image](https://github.com/user-attachments/assets/90136685-e137-4753-816e-21ea6eb59d1e)

**Comment by sailro** [comment]
^cc @MichalPavlik

**Comment by julienp** [comment]
> This also currently breaks .NET custom providers in pulumi:

Indeed, we just ran into this when trying to add tests for .NET 9.0 in Pulumi's CI. As a workaround we'll strip the control characters https://github.com/pulumi/pulumi/pull/17783

I'd love to see an upstream fix for this in the next release.
